.github/workflows/get-environment.yml

```diff
@@ -259,11 +259,24 @@ jobs:
 
       - name: "Get release: 1 for testing / stable, <date>.<commit_sha> for others"
         id: get_release
-        run: |
-          RELEASE=$(date '+%H%M%S')
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            let release = '';
+
+            if (${{ steps.get_stability.outputs.stability }} === 'testing') {
+              const branchName = "${{ github.ref_name }}";
+              const matches = branchName.match(/^(?:release|hotfix)-(\d{8})$/);
+              if (matches) {
+                release = matches[1];
+              } else {
+                throw new Error('invalid release name');
+              }
+            } else {
+              release = new Date().toLocaleTimeString('en-US', { hour12: false }).replaceAll(":","")
```

Comment:
```suggestion
              const currentDate = new Date();
              release = `${currentDate.getFullYear()}${("0" + (currentDate.getMonth() + 1)).slice(-2)}00`;
```

Reply:
i'm not sure about it, the `release` variable should be defined in the same vein as it used to be (`RELEASE=$(date '+%H%M%S')`) when not in a testing/release context